security/advancedtls/advancedtls_test.go

```diff
@@ -679,3 +681,83 @@ func TestOptionsConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestGetCertificateSNI(t *testing.T) {
+	// Load server certificates for setting the serverGetCert callback function.
+	serverPeerCert1, err := tls.LoadX509KeyPair(testdata.Path("server_cert_1.pem"),
+		testdata.Path("server_key_1.pem"))
+	if err != nil {
+		t.Fatalf("Server is unable to parse peer certificates. Error: %v", err)
+	}
+	serverPeerCert2, err := tls.LoadX509KeyPair(testdata.Path("server_cert_2.pem"),
+		testdata.Path("server_key_2.pem"))
+	if err != nil {
+		t.Fatalf("Server is unable to parse peer certificates. Error: %v", err)
+	}
+	serverPeerCert3, err := tls.LoadX509KeyPair(testdata.Path("server_cert_3.pem"),
+		testdata.Path("server_key_3.pem"))
+	if err != nil {
+		t.Fatalf("Server is unable to parse peer certificates. Error: %v", err)
+	}
+	tests := []struct {
+		desc                string
+		serverGetCert       func(*tls.ClientHelloInfo) ([]*tls.Certificate, error)
+		serverName          string
+		expectedCertificate tls.Certificate
+	}{
+		{
+			desc: "Selected certificate by SNI should be serverPeerCert1",
```

Comment:
The `desc` field which is passed as the first argument to `t.Run` is used to generate the Go subtest name. So, this subtest would get the following name: `TestGetCertificateSNI/Selected_certificate_by_SNI_should_be_serverPeerCert1`. So, instead of using complete sentences here, think more of these in terms of identifier names and use short and descriptive names. Thanks.

Reply:
Will do!